### PR TITLE
Cpu support

### PR DESCRIPTION
**Pull request by HonzaCuhel**

### CODE REVIEW

File: src/datadreamer/image_generation/sdxl_image_generator.py [hunk 1]
```diff
@@ -37,22 +37,40 @@ def _init_gen_model(self):
...
-        )
-        refiner.enable_model_cpu_offload()
+        if self.device == "cpu":
+            base = DiffusionPipeline.from_pretrained(
+                "stabilityai/stable-diffusion-xl-base-1.0",
+                # variant="fp16",
+                torch_dtype=torch.float32,
+                use_safetensors=True,
+            )
+            base.to("cpu")
```
Comment (tersekmatija) [inline src/datadreamer/image_generation/sdxl_image_generator.py:47]: I think this block should be simplified by adding the if statement here directly. Same for below.

Comment (HonzaCuhel) [reply src/datadreamer/image_generation/sdxl_image_generator.py:47]: Well, in the CUDA branch, not only the `torch_dtype` is different, but there are `variant` arguments specified, and after the models are also initialized, `base.enable_model_cpu_offload()` is called, none of which is used in the CPU branch. Would you still like me to put the if directly inside, @tersekmatija?

File: src/datadreamer/image_generation/clip_image_tester.py [hunk 2]
```diff
@@ -39,11 +42,13 @@ def test_image(self, image: Image.Image, objects: List[str], conf_threshold=0.05
...
         inputs = self.clip_processor(
             text=objects, images=image, return_tensors="pt", padding=True
-        )
+        ).to(self.device)
 
         # Get similarity scores from the CLIP model
         outputs = self.clip(**inputs)
-        logits_per_image = outputs.logits_per_image  # image-text similarity score
+        logits_per_image = (
+            outputs.logits_per_image.detach().cpu()
```
Comment (kozlov721) [inline src/datadreamer/image_generation/clip_image_tester.py:50]: should the `.cpu()` be here?

File: src/datadreamer/image_generation/sdxl_image_generator.py [hunk 3]
```diff
@@ -37,22 +37,40 @@ def _init_gen_model(self):
...
+            refiner = DiffusionPipeline.from_pretrained(
+                "stabilityai/stable-diffusion-xl-refiner-1.0",
+                text_encoder_2=base.text_encoder_2,
+                vae=base.vae,
+                torch_dtype=torch.float32,
+                use_safetensors=True,
+                # variant="fp16",
```
Comment (kozlov721) [inline src/datadreamer/image_generation/sdxl_image_generator.py:54]: Why is the `variant` sometimes commented out and sometimes not?

File: src/datadreamer/image_generation/sdxl_turbo_image_generator.py [hunk 4]
```diff
@@ -32,13 +32,22 @@ def _init_gen_model(self):
...
-            "stabilityai/sdxl-turbo",
-            torch_dtype=torch.float16,
-            variant="fp16",
-            use_safetensors=True,
-        )
-        base.enable_model_cpu_offload()
+        if self.device == "cpu":
+            base = AutoPipelineForText2Image.from_pretrained(
+                "stabilityai/sdxl-turbo",
+                # variant="fp16",
```
Comment (kozlov721) [inline src/datadreamer/image_generation/sdxl_turbo_image_generator.py:38]: The same question about `variant` as above.

File: src/datadreamer/image_generation/sdxl_image_generator.py (hunk 1 above)
Comment (sokovninn) [reply src/datadreamer/image_generation/sdxl_image_generator.py:47]: Agree with the current approach, because of `base.enable_model_cpu_offload()` and `variant`.

File: src/datadreamer/image_generation/clip_image_tester.py (hunk 2 above)
Comment (sokovninn) [reply src/datadreamer/image_generation/clip_image_tester.py:50]: @HonzaCuel why did you add .detach().cpu()? Shouldn't it work without this?

File: src/datadreamer/image_generation/sdxl_turbo_image_generator.py (hunk 4 above)
Comment (sokovninn) [reply src/datadreamer/image_generation/sdxl_turbo_image_generator.py:38]: Because it's not supported for the cpu. Right @HonzaCuhel ?

File: src/datadreamer/image_generation/clip_image_tester.py (hunk 2 above)
Comment (HonzaCuhel) [reply src/datadreamer/image_generation/clip_image_tester.py:50]: @sokovninn, yeah, you're probably right; I'll try it.

File: src/datadreamer/image_generation/sdxl_turbo_image_generator.py (hunk 4 above)
Comment (HonzaCuhel) [reply src/datadreamer/image_generation/sdxl_turbo_image_generator.py:38]: Yeah, fp16 is supported only on GPUs, not CPUs.

**Review by sokovninn — APPROVED** [review]
Looks good now
